rust/src/nasl/utils/error.rs

```diff
@@ -3,136 +3,159 @@
 // SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
 
 //! Defines function error kinds
-use std::io;
 use thiserror::Error;
 
+use crate::nasl::builtin::{BuiltinError, SshError};
 use crate::nasl::prelude::NaslValue;
 
 use crate::storage::StorageError;
 
-use super::super::builtin::SshError;
 use super::ContextType;
 
 /// Reuses the StorageError definitions as they should fit most cases.
 pub type GeneralErrorType = StorageError;
 
 #[derive(Debug, Clone, PartialEq, Eq, Error)]
-/// Descriptive kind of error that can occur while calling a function
-pub enum FunctionErrorKind {
-    /// Function called with insufficient arguments
+pub enum ArgumentError {
     #[error("Expected {expected} but got {got}")]
-    MissingPositionalArguments {
-        /// Expected amount of arguments
-        expected: usize,
-        /// Actual amount of arguments
-        got: usize,
-    },
-    /// Function called with trailing positional arguments
+    MissingPositionals { expected: usize, got: usize },
     #[error("Expected {expected} but got {got}")]
-    TrailingPositionalArguments {
-        /// Expected amount of arguments
-        expected: usize,
-        /// Actual amount of arguments
-        got: usize,
-    },
-    /// Function called without required named arguments
+    TrailingPositionals { expected: usize, got: usize },
     #[error("Missing arguments: {}", .0.join(", "))]
-    MissingArguments(Vec<String>),
-    /// Function called with additional, unexpected named arguments
+    MissingNamed(Vec<String>),
     #[error("Unknown named argument given to function: {}", .0)]
     UnexpectedArgument(String),
-    /// Wraps formatting error
-    #[error("Formatting error: {0}")]
-    FMTError(#[from] std::fmt::Error),
-    /// Wraps io::Error
-    #[error("IOError: {0}")]
-    IOError(io::ErrorKind),
-    /// Function was called with wrong arguments
     #[error("Function was called with wrong arguments: {0}")]
     WrongArgument(String),
-    /// Authentication failed
-    #[error("Authentication failed.")]
-    Authentication,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Error)]
+pub enum InternalError {
+    #[error("{0}")]
+    Storage(#[from] StorageError),
+    #[error("{0}")]
+    Dirty(String),
```

Comment:
can we get rid of Dirty? Dirty in StorageError is used because it is short version of dirty data which includes but is not limited to corruption or undefined behaviour. I don't think it makes sense as an InternalError. I think we should be more considered and may define extra cases that are not Storage specific.

Reply:
Yes, I just wasnt finished and dirty is used a ton. Got rid of it in some of the newer commits.